scipy/stats/tests/test_quantile.py

```diff
@@ -410,3 +415,342 @@ def test_nd(self, side, ties, shape, nans_x, infs_x, xp):
         x, y = xp.asarray(x), xp.asarray(y)
         res = _xp_searchsorted(x, y, side=side)
         xp_assert_equal(res, ref)
+
+
+@_apply_over_batch(('x', 1), ('y', 1))
+def ogive_reference_last_axis(x, y, nan_policy, method):
+    i_nan = np.isnan(x)
+    if nan_policy == 'propagate' and np.any(i_nan):
+        return np.full_like(y, np.nan)
+    elif nan_policy == 'omit':
+        x = x[~i_nan]
+    return stats.ogive(x, y, keepdims=True, method=method)
+
+
+def ogive_reference(x, y, *, axis=0, nan_policy='propagate',
+                        keepdims=None, method='linear'):
+    x, y = _broadcast_arrays((x, y), axis=axis)
+    x, y = np.moveaxis(x, axis, -1), np.moveaxis(y, axis, -1)
+    res = ogive_reference_last_axis(x, y, nan_policy, method)
+    res = np.moveaxis(res, -1, axis)
+    if not keepdims:
+        res = np.squeeze(res, axis=axis)
+    return res
+
+
+_ogive_methods_list = sorted(list(_ogive_methods))  # avoid variable collection order
+
+
+@make_xp_test_case(stats.ogive)
+class TestOgive:
+    def test_input_validation(self, xp):
+        x = xp.asarray([1, 2, 3])
+        y = xp.asarray(2)
+
+        message = "`x` must have real dtype."
+        with pytest.raises(ValueError, match=message):
+            stats.ogive(xp.asarray([True, False]), y)
+        with pytest.raises(ValueError):
+            stats.ogive(xp.asarray([1+1j, 2]), y)
+
+        message = "`y` must have real dtype."
+        with pytest.raises(ValueError, match=message):
+            stats.ogive(x, xp.asarray([0+1j, 1]))
+
+        message = "`axis` must be an integer or None."
+        with pytest.raises(ValueError, match=message):
+            stats.ogive(x, y, axis=0.5)
+        with pytest.raises(ValueError, match=message):
+            stats.ogive(x, y, axis=(0, -1))
+
+        message = "`axis` is not compatible with the shapes of the inputs."
+        with pytest.raises(ValueError, match=message):
+            stats.ogive(x, y, axis=2)
+
+        if not is_jax(xp):  # no data-dependent input validation for lazy arrays
+            message = "The input contains nan values"
+            with pytest.raises(ValueError, match=message):
+                stats.ogive(xp.asarray([xp.nan, 1, 2]), y, nan_policy='raise')
+
+        message = "method` must be one of..."
+        with pytest.raises(ValueError, match=message):
+            stats.ogive(x, y, method='a duck')
+
+        message = "If specified, `keepdims` must be True or False."
+        with pytest.raises(ValueError, match=message):
+            stats.ogive(x, y, keepdims=42)
+
+        message = "`keepdims` may be False only if the length of `y` along `axis` is 1."
+        with pytest.raises(ValueError, match=message):
+            stats.ogive(x, xp.asarray([0.5, 0.6]), keepdims=False)
+
+    @pytest.mark.parametrize('method', _ogive_methods_list)
+    @pytest.mark.parametrize('dtype', [None, 'float32', 'float64'])
```

Comment:
The simplest way to speed up the tests without much loss of coverage is to remove `dtype` parameterization here. I think it's sufficient to `parametrize` over `dtype` in `test_against_reference` only.